.pre-commit-config.yaml

```
@@ -1,14 +1,14 @@
 repos:
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.11.8
+    rev: v0.12.8
     hooks:
       - id: ruff
         name: Run Ruff (lint) on Doc/
         args: [--exit-non-zero-on-fix]
         files: ^Doc/
       - id: ruff
         name: Run Ruff (lint) on Lib/test/
-        args: [--exit-non-zero-on-fix]
+        args: [--exit-non-zero-on-fix, --target-version=py313]
```

I guess a low version was initially chosen to make backports easier?

Anyway, let's move the config to a `.ruff.toml` file so it also applies if running Ruff directly.


```suggestion
        args: [--exit-non-zero-on-fix]
```

We have a top-level that defines `target-version = "py310"`.

Let's override that in `Lib/test/.ruff.toml`, either:

* `target-version = "py313"` like in `Doc/.ruff.toml`
* `[per-file-target-version]` like in `Tools/build/.ruff.toml` (might be too files many to be worth it)

And we can also remove the t-string files from `extend-exclude` and put those under `[per-file-target-version]` for `py314`.

(Could do the same for `test_grammar.py` but it has some F811 errors that would need deciding about.)

---

Done.